### PR TITLE
Add click link ui functions

### PR DESCRIPTION
**Pull request by avh4**
Depends on #8.  This PR starts with 21f8664

### CODE REVIEW

File: src/Navigation/Extra.elm [hunk 1]
```diff
@@ -38,4 +38,8 @@ resolve : Navigation.Location -> String -> Navigation.Location
 resolve base url =
     locationFromString url
         -- TODO: implment correct logic (current logic is only correct for "authority-relative" URLs without query or fragment strings)
-        |> Maybe.withDefault { base | pathname = url }
+        |> Maybe.withDefault
+            { base
+                | href = (base.href |> String.split "/" |> List.take 3 |> String.join "/") ++ url
```
Comment (BrianHicks) [inline src/Navigation/Extra.elm]: is this to select only `http://blah.com/`? A comment might be useful there, but this is a nit.

File: src/TestContext.elm [hunk 2]
```diff
@@ -11,16 +14,20 @@ module TestContext
...
+        , fillInTextarea
         , routeChange
         , shouldHave
         , shouldHaveLastEffect
         , shouldHaveView
         , shouldNotHave
         , simulate
         , update
+        , within
         )
```
Comment (jamesgary) [inline src/TestContext.elm]: FYI, I opened a [PR](https://github.com/NoRedInk/NoRedInk/pull/19554) last night that took functions like `clickOn` and `ifThenElse` from `ui/tests/Page/Teach/Assignment/Form/UiActions.elm` and put them in a new file `ui/tests/SpecHelpers/UiActions.elm`. However, I think putting the helper functions in this file makes perfect sense, so once your PR is merged, I can move the `UiActions` functions into this file, if that sounds OK to you?

File: src/TestContext.elm [hunk 3]
```diff
@@ -536,28 +536,67 @@ clickLink linkText href testContext =
...
-                            -- there is a click handler, so simulate that event and ignore the `href`
+                    let
+                        link =
+                            program.view model
+                                |> findLinkTag
+                    in
+                    if respondsTo normalClick link then
+                        -- there is a click handler
+                        -- first make sure the handler properly respects "Open in new tab", etc
+                        if respondsTo ctrlClick link || respondsTo metaClick link then
```
Comment (BrianHicks) [inline src/TestContext.elm]: wow, this is super cool. Thank you for making sure this behaves nicely <3

File: src/Navigation/Extra.elm (hunk 1 above)
Comment (avh4) [reply src/Navigation/Extra.elm]: The relevant comment here, I think, is the one at the top of the module
> this module should implement the algorithm described at <https://url.spec.whatwg.org/>
:)

**Pull request merged**